src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -223,13 +229,47 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
             builder.field(fieldName, schema);
             break;
           }
+        } else if (mapNumerics == NumericMapping.COMPLETE) {
```

Comment:
Adding the implementation in a similar block here rather than consolidating potentially to a method. I'd prefer to keep the changes focused on this PR. There can be a follow up PR to make things more compact later.